backend/apps/user/src/database/migrations/20230921061446_create-user-profile-table.ts

```diff
@@ -0,0 +1,18 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.createTable('userProfiles', function (table) {
+    table.increments('id').primary();
+    table.uuid('userId').notNullable(); // references 'id' column in the 'users' table in the auth microservice
```

Comment:
I was reading up on how to maintain foreign relations on tables in different DBs, and seems like the standard way of dealing with FK constraints in different microservice databases is to use our messaging queue, so that each microservice can consume the messages they want to and make changes to their database accordingly.

I think for our case, both the auth and user microservice need to consume messages that have to do with user creation and deletion, e.g. when a find_or_create_user message is sent from gateway, both the auth and user microservice will consume it and make changes to the databases accordingly.

Can refer to this link: https://stackoverflow.com/questions/62920993/how-to-deal-with-foreign-key-in-microservice-architecture

Reply:
Resolved by merging auth service into user service since the 2 have many overlapping domain definitions and will be highly coupled if they are in separate microservices